AutoRuns.psm1

```diff
@@ -2160,30 +2165,38 @@ Begin {
                 If ($VerifyDigitalSignature) {
                     if ($_.ImagePath) {
                         If (Test-Path -Path $_.ImagePath -PathType Leaf) {
-                            $_ | Add-Member -MemberType ScriptProperty -Name Signed -Value ({
-                                try {
-                                    $signature = Get-AuthenticodeSignature -FilePath $($this.ImagePath) -ErrorAction Stop
-                                    Switch ($signature.Status) {
-                                        'Valid' {
-                                            $true
-                                            break
-                                        }
-                                        'NotSigned' {
-                                            $false
-                                            break
-                                        }
-                                        default {
-                                            $false
-                                        }
-                                    }
-
-                                } catch {
+                            
+                            ## Add the signature status to the entry
+                            $signature = Get-AuthenticodeSignature -FilePath $_.ImagePath -ErrorAction Stop
+                            $signed = switch ($signature.Status) {
+                                'Valid' {
+                                    $true
+                                    break
+                                }
+                                'NotSigned' {
+                                    $false
+                                    break
+                                }
+                                default {
                                     $false
                                 }
-                            }) -Force -PassThru
+                            }
+                            $_ = $_ | Add-Member -MemberType NoteProperty -Name Signed -Value $signed -Force -PassThru
```

Comment:
Just so that the pipeline doesn't get too long and hard to read. Also, it is sometimes in a conditional, which makes it even harder to understand as a pipeline.